tutorials/rag/agentic_rag.py

```diff
@@ -0,0 +1,274 @@
+"""
+This script demonstrates how to build a more agentic RAG pipeline using Humanloop and OpenAI's API.
+
+The model can choose to call the retrieval tool, or instead ask for clarification from the user.
+
+This leverages tool calling: https://platform.openai.com/docs/guides/function-calling
+
+It assumes that Prompts and Tools are managed in the client's code and logging and evals are done via Humanloop.
+"""
+
+import json
+from dotenv import load_dotenv
+import os
+from chromadb import chromadb
+from openai import OpenAI
+import pandas as pd
+from datetime import datetime
+from humanloop import Humanloop
+
+# Load .env file that contains API keys
+load_dotenv()
+
+# Init clients
+hl = Humanloop(api_key=os.getenv("HUMANLOOP_KEY"))
+chroma = chromadb.Client()
+openai = OpenAI(api_key=os.getenv("OPENAI_KEY"))
```

Comment:
also flagging that this os.getenv is very unnecessary and doesn't conform to the standard `OPENAI_API_KEY` and `HUMANLOOP_API_KEY` env vars that i think the openai and humanloop clients will both automatically pick up.

unfortunately all the guides will have to updated for this env var to be corrected.

Reply:
should be an easy change